manifest/protobuf_test.go

```diff
@@ -349,3 +350,255 @@ func TestReaderOptions_ProtoPathAndDescriptorSet(t *testing.T) {
+
+	// Create realistic test data similar to what buf.build would return
+	testFds := &descriptorpb.FileDescriptorSet{
+		File: []*descriptorpb.FileDescriptorProto{
+			{
+				Name:    proto.String("google/protobuf/descriptor.proto"),
+				Package: proto.String("google.protobuf"),
+				MessageType: []*descriptorpb.DescriptorProto{
+					{
+						Name: proto.String("FileDescriptorSet"),
+						Field: []*descriptorpb.FieldDescriptorProto{
+							{
+								Name:   proto.String("file"),
+								Number: proto.Int32(1),
+								Type:   descriptorpb.FieldDescriptorProto_TYPE_MESSAGE.Enum(),
+								TypeName: proto.String(".google.protobuf.FileDescriptorProto"),
+								Label:    descriptorpb.FieldDescriptorProto_LABEL_REPEATED.Enum(),
+							},
+						},
+					},
+				},
+			},
+			{
+				Name:    proto.String("sf/substreams/sink/service/v1/service.proto"),
+				Package: proto.String("sf.substreams.sink.service.v1"),
+				MessageType: []*descriptorpb.DescriptorProto{
+					{
+						Name: proto.String("DeployRequest"),
+						Field: []*descriptorpb.FieldDescriptorProto{
+							{
+								Name:   proto.String("substreams_package"),
+								Number: proto.Int32(1),
+								Type:   descriptorpb.FieldDescriptorProto_TYPE_MESSAGE.Enum(),
+								TypeName: proto.String(".sf.substreams.v1.Package"),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Generate cache key for realistic module/version/symbols
+	module := "buf.build/streamingfast/substreams-sink-sql"
+	version := "v1.0.0"
+	symbols := []string{
+		"sf.substreams.sink.service.v1.DeployRequest",
+		"sf.substreams.sink.service.v1.DeployResponse",
+	}
+	cacheKey := generateCacheKey(module, version, symbols)
+
+	// Test: First call should save to cache
+	err := saveToCache(cacheKey, testFds)
+	if err != nil {
+		t.Fatalf("Failed to save realistic data to cache: %v", err)
+	}
+
+	// Test: Second call should load from cache
+	cachedFds, err := loadFromCache(cacheKey)
+	if err != nil {
+		t.Fatalf("Failed to load realistic data from cache: %v", err)
+	}
+
+	// Verify integrity of cached data
+	if len(cachedFds.File) != len(testFds.File) {
+		t.Errorf("Cache integrity check failed: expected %d files, got %d", len(testFds.File), len(cachedFds.File))
+	}
+
+	// Verify specific file content
+	for i, originalFile := range testFds.File {
+		cachedFile := cachedFds.File[i]
+		
+		if cachedFile.GetName() != originalFile.GetName() {
+			t.Errorf("File name mismatch at index %d: expected %s, got %s", 
+				i, originalFile.GetName(), cachedFile.GetName())
+		}
+		
+		if cachedFile.GetPackage() != originalFile.GetPackage() {
+			t.Errorf("Package name mismatch at index %d: expected %s, got %s", 
+				i, originalFile.GetPackage(), cachedFile.GetPackage())
+		}
+
+		if len(cachedFile.MessageType) != len(originalFile.MessageType) {
+			t.Errorf("Message type count mismatch at index %d: expected %d, got %d", 
+				i, len(originalFile.MessageType), len(cachedFile.MessageType))
+		}
```

Comment:
`proto.Equal` needs to be used to determine protobuf equality. Usually we couple that with a `require.EqualExported` (not sure of the exact name but check equality only on exported fields of a struct) which check correct equality rules and print a diff if it's not good:

```
if !proto.Equal(a, b) {
  require.EqualExported(t, a, b) // Fails and print diff
}
```